tableauserverclient/models/interval_item.py

```diff
@@ -0,0 +1,175 @@
+import xml.etree.ElementTree as ET
+from datetime import datetime
+from .. import NAMESPACE
+
+
+def _validate_time(t):
+    units_of_time = {"hour", "minute", "second"}
+
+    if not any(hasattr(t, unit) for unit in units_of_time):
+        error = "Invalid time object defined."
+        raise ValueError(error)
+
+
+class IntervalItem(object):
+    class Frequency:
+        Hourly = "Hourly"
+        Daily = "Daily"
+        Weekly = "Weekly"
+        Monthly = "Monthly"
+
+    class Occurrence:
+        Hours = "hours"
+        Minutes = "minutes"
+        WeekDay = "weekDay"
+        MonthDay = "monthDay"
+
+    class Day:
+        Sunday = "Sunday"
+        Monday = "Monday"
+        Tuesday = "Tuesday"
+        Wednesday = "Wednesday"
+        Thursday = "Thursday"
+        Friday = "Friday"
+        Saturday = "Saturday"
+        LastDay = "LastDay"
+
+
+class HourlyInterval(object):
+    def __init__(self, start_time, end_time, interval_occurrence, interval_value):
+
+        self.start_time = start_time
```

Comment:
This feels cleaner to me because it hide the _blah versions as the implementation detail that they are and keeps the `__init__` function clean.

I then don't need to have a bunch of these in `__init__`:
```
        # Invoke setter
        self.name = name
```